shm/Map.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Map.h"
+
+#include <algorithm>
+#include <random>
+
+#include "Island.h"
+
+Map::Map() {
+    for (int i = 0; i < islandsCount; i++) {
+        std::random_device rd;
+        std::mt19937 gen(rd());
```

Comment:
I think this might be instantiated outside the loop

Reply:
fixed